torch/distributed/distributed_c10d.py

```diff
@@ -1411,6 +1441,10 @@ def _new_process_group_helper(
 
         pg._register_backend(torch.device(device), backend_type, backend_class)
 
+    if pg.supports_splitting and device_id:
+        eager_backend = pg._get_backend(device_id)
+        eager_backend.eager_connect_single_device(device_id)
+
```

Comment:
nit: we could consider decoupling `support_splitting` and early connect. i.e. a backend may support early connect but not splitting.